zrml/prediction-markets/src/tests.rs

```diff
@@ -2491,8 +2499,10 @@ fn reject_market_fails_on_approved_market() {
         // Creates an advised market.
         simple_create_categorical_market(MarketCreation::Advised, 0..1, ScoringRule::CPMM);
         assert_ok!(PredictionMarkets::approve_market(Origin::signed(SUDO), 0));
+        let reject_reason: RejectReason =
+            vec![0; 1024].try_into().expect("BoundedVec conversion failed");
         assert_noop!(
-            PredictionMarkets::reject_market(Origin::signed(SUDO), 0),
+            PredictionMarkets::reject_market(Origin::signed(SUDO), 0, reject_reason),
             Error::<Runtime>::InvalidMarketStatus
         );
     });
```

Comment:
We don't have that yet but I think it would be nice to see that events are thrown properly, i.e. that the rejection event contains all expected elements. Would also be nice to see that the extrinsic fails when the reason is too long.

[Here is some example to check events](https://github.com/paritytech/substrate/blob/0ee03277c33b6334ddba7434e014fa637dcb6107/frame/balances/src/tests.rs#L733-L756) or in case you want to reset event history and check the order of all new events, [take a look at this snippet](https://github.com/paritytech/substrate/blob/5a8f3e0a9cb45198b67c4eeba20062085a9381ce/frame/balances/src/tests.rs#L758-L852)